Tools/LocomotionTool/LocomotionTool.cs

```diff
@@ -0,0 +1,559 @@
+			{
+				this.UnlockRay(rayOrigin, this);
+				this.SetDefaultRayVisibility(rayOrigin, true);
+
+				if (!m_BlinkVisuals.outOfMaxRange)
+				{
+					m_BlinkVisuals.HideVisuals();
+					StartCoroutine(MoveTowardTarget(m_BlinkVisuals.locatorPosition));
+				}
+				else
+				{
+					m_BlinkVisuals.enabled = false;
+					m_State = State.Inactive;
+				}
+			}
+		}
+
+		void DoTwoHandedScaling(ConsumeControlDelegate consumeControl)
+		{
+			if (this.IsSharedUpdater(this))
+			{
+				var grip = m_LocomotionInput.grip;
+				if (grip.isHeld)
+				{
+					if (m_AllowScaling)
+					{
+						var otherGripHeld = false;
+						foreach (LocomotionTool locomotionTool in linkedObjects)
+						{
+							if (locomotionTool == this)
+								continue;
+
+							var otherLocomotionInput = locomotionTool.m_LocomotionInput;
+							var otherGrip = otherLocomotionInput.grip;
+							if (otherGrip.isHeld)
+							{
+								otherGripHeld = true;
+								consumeControl(grip);
+								consumeControl(otherGrip);
+
+								var thisPosition = cameraRig.InverseTransformPoint(rayOrigin.position);
+								var otherRayOrigin = locomotionTool.rayOrigin;
+								var otherPosition = cameraRig.InverseTransformPoint(otherRayOrigin.position);
+								var distance = Vector3.Distance(thisPosition, otherPosition);
+
+								var rayToRay = otherPosition - thisPosition;
+								var midPoint = thisPosition + rayToRay * 0.5f;
+
+								rayToRay.y = 0; // Use for yaw rotation
+
+								var pivotYaw = MathUtilsExt.ConstrainYawRotation(cameraRig.rotation);
+
+								if (!m_Scaling)
+								{
+									m_StartScale = this.GetViewerScale();
+									m_StartDistance = distance;
+									m_StartMidPoint = pivotYaw * midPoint * m_StartScale;
+									m_StartPosition = cameraRig.position;
+									m_StartDirection = rayToRay;
+									m_StartYaw = cameraRig.rotation.eulerAngles.y;
+
+									locomotionTool.m_Scaling = true;
+
+									CreateViewerScaleVisuals(rayOrigin, otherRayOrigin);
+								}
+
+								m_Scaling = true;
+
+								var currentScale = Mathf.Clamp(m_StartScale * (m_StartDistance / distance), k_MinScale, k_MaxScale);
+
+								var thumb = m_LocomotionInput.thumb;
+								var thumbHeld = thumb.isHeld;
+								if (thumbHeld)
+									consumeControl(thumb);
+
+								var otherThumb = otherLocomotionInput.thumb;
+								var otherThumbHeld = otherThumb.isHeld;
+								if (otherThumbHeld)
+									consumeControl(otherThumb);
+
+								// Press both thumb buttons to reset scale
+								if (thumbHeld && otherThumbHeld)
+								{
+									m_AllowScaling = false;
+
+									rayToRay = otherRayOrigin.position - rayOrigin.position;
+									midPoint = rayOrigin.position + rayToRay * 0.5f;
+									var currOffset = midPoint - cameraRig.position;
+
+									cameraRig.position = midPoint - currOffset / currentScale;
+									cameraRig.rotation = Quaternion.AngleAxis(m_StartYaw, Vector3.up);
```

Comment:
The idea is to undo what you just did with the two-handed scale, since you might just be wanting to reset your scale. @timoni, thoughts?